code/modules/materials/Mat_Materials.dm

```diff
@@ -257,12 +451,85 @@ ABSTRACT_TYPE(/datum/material)
 			X.execute(owner, attackobj, attacker, meleeorthrow)
 		return
 
-
+//Material definitions
 /datum/material/interpolated
 	mat_id = "imcoderium"
 	name = "imcoderium"
 	desc = "You should not be seeing this"
 	color = "#6f00ff"
+	cached = FALSE
+
+	///Create an interpolated material from two input materials, with bias. Bias of 0 is entirely mat1, bias of 1 is entirely mat2
+	New(var/datum/material/mat1,var/datum/material/mat2,var/bias)
+		..()
+		if(isnull(mat1) || isnull(mat2))
+			CRASH("Tried to create alloy with null materials!")
+		var/left_bias = 1 - bias
+		src.quality = round(mat1.quality *left_bias+ mat2.quality * bias)
+
+		src.prefixes = (mat1.prefixes | mat2.prefixes)
+		src.suffixes = (mat1.suffixes | mat2.suffixes)
+
+		src.value = round(mat1.value *left_bias+ mat2.value * bias)
+		src.name = mat1.interpolateName(mat2, 0.5)
+		src.desc = "This is an alloy of [mat1.name] and [mat2.name]"
+		src.mat_id = "([mat1.getID()]+[mat2.getID()])"
+		src.alpha = round(mat1.alpha *left_bias+ mat2.alpha * bias)
+		if(islist(mat1.color) || islist(mat2.color))
+			var/list/colA = normalize_color_to_matrix(mat1.color)
+			var/list/colB = normalize_color_to_matrix(mat2.color)
+			src.color = list()
+			for(var/i in 1 to length(colA))
+				src.color += colA[i] *left_bias+ colB[i] * bias
+		else
+			src.color = rgb(round(GetRedPart(mat1.color) *left_bias+ GetRedPart(mat2.color) * bias), round(GetGreenPart(mat1.color) *left_bias+ GetGreenPart(mat2.color) * bias), round(GetBluePart(mat1.color) *left_bias+ GetBluePart(mat2.color) * bias))
+		src.properties = mergeProperties(mat1.properties, mat2.properties, bias)
+
+		src.edible_exact = round(mat1.edible_exact *left_bias+ mat2.edible_exact * bias)
```

Comment:
Won't this just always equal `0` or `1` since 1-argument `round` just floors the number?
Should probably look something like
```
src.edible_exact = [calculation]
src.edible = round(src.edible_exact, 1)
```
(yes, `round(n)` and `round(n, 1)` behave differently for some reason. why)

Reply:
I'm not entirely sure why the round is there anyway. I wonder if this has just not been working since alloying was put in.